### DISCUSSION TITLE
Something WAY down the line. Self-learning controls.

### DISCUSSION THREAD

**Discussion by Jonathanese** [Rem0o/FanControl.Releases#674] [Ideas]
This would be a ways off and something I might look into creating my own implementation for. But if it's something someone can get working, it would make FanControl pretty much run several laps around proprietary software.

Basically the idea would be this:
Select temperatures you want to monitor.
Select ideal temperatures.
Select fans that may be relevant.

From there, it'll set up a baseline control. _However_, over time as it retrieves more of a range of values on the sensors and fans, it can tweak this baseline control based on what it learns.

So for instance, it might start off by blasting the fans when the CPU or GPU is hot. But over time, it might learn that running the CPU fan also reduces ambient temps and can run the GPU fans at a lower speed. Or it might learn that running the GPU at a higher speed has more of an affect on GPU temps than running the rear exhaust fan, so it can emphasize faster GPU fans.

The idea is that these behaviors and correlations wouldn't be explicitly defined. It would just have a slow and small neural net using sensors at the inputs and fan speeds as the outputs, and fitness values based on ideal ranges and minimum fan speeds, and through normal usage, it could create its own correlations. The more sensors you give it, more situations you give it, and longer you give it, the the better it gets at keeping your system cool and quiet.

**Comment by Rem0o** [comment]
You can take a look at the Auto fan curve. It's my first attempt at something in this direction.

**Reply by Jonathanese** [reply]
I'll check it out!
